packages/flutter_tools/lib/src/asset.dart

```diff
@@ -1285,8 +1285,16 @@ class _AssetDirectoryCache {
   List<String> variantsFor(String assetPath) {
     final String directory = _fileSystem.path.dirname(assetPath);
 
-    if (!_fileSystem.directory(directory).existsSync()) {
-      return const <String>[];
+    try {
```

Comment:
can we catch this at the file, and not directory level?

Reply:
per andrew, not a blocker.

Reply:
I learned that that `existsSync` only throws this exception in the scenario where entity in question is a directory (not a file).